### ISSUE TITLE
Seadrive continuously tries to sync a file

### ISSUE THREAD

**Issue by didrip** [haiwen/seadrive-gui#128]
Upload/Download of every file is really slow on a machine running macOS12 and seadrive 1.0.3, it used to be OK.
No network problem whatsoever, transfer speed via other means (HTTP, SMB...) is good.

Looking at `seadrive.log` it seems that seadrive is "blocking" on an unexisting file with weird path : `/Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com`

https://seadrive.server.com is the URL of our instance of Seafile server and user@email.com is the login used to connect on this machine (macbook).

The file at specified path /Users/username/.seadrive/data/file-cache/root/ does not exists.

`seadrive.log` :

```
[01/29/19 14:21:01] file-cache-mgr.c(3236): Failed to open dir /Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com: Error opening directory “/Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com”: No such file or directory.
[01/29/19 14:21:03] file-cache-mgr.c(3236): Failed to open dir /Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com: Error opening directory “/Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com”: No such file or directory.
[01/29/19 14:21:04] file-cache-mgr.c(3236): Failed to open dir /Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com: Error opening directory “/Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com”: No such file or directory.
[01/29/19 14:21:06] file-cache-mgr.c(3236): Failed to open dir /Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com: Error opening directory “/Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com”: No such file or directory.
[01/29/19 14:21:08] file-cache-mgr.c(3236): Failed to open dir /Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com: Error opening directory “/Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com”: No such file or directory.
[01/29/19 14:21:09] file-cache-mgr.c(3236): Failed to open dir /Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com: Error opening directory “/Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com”: No such file or directory.
[01/29/19 14:21:11] file-cache-mgr.c(3236): Failed to open dir /Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com: Error opening directory “/Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com”: No such file or directory.
[01/29/19 14:21:13] file-cache-mgr.c(3236): Failed to open dir /Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com: Error opening directory “/Users/username/.seadrive/data/file-cache/root/https___seadrive.server.com_user@email.com”: No such file or directory.
```

and it goes on and on

Logging out with clearing cache and logging back in fixes the issue but only temporarily, it comes back after some time.

- Seadrive 1.0.3
- Seadrive server Pro 6.3.12
- MacOS 10.12

**Comment by monkeydri** [comment]
Issue remains with Seadrive 1.0.4.

This time I even uninstalled Seadrive and deleted the .seadrive folder completely. 

Actually looking at it more closely clearing the cache does not solve the issue even temporarily, error messages comes back straight after login.

**Comment by killing** [comment]
The log you see is not relevant to "slow" problem. How slow is upload/download? What network do you use for testing? Do you have the same speed when you use the sync client or web interface?

**Comment by monkeydri** [comment]
When a file is added to a synced library, it takes at least one minute to appear on the problematic mac machine, even for a small file of few KB.

Upload/download speed (visible in Transfer progress in context menu if Seadrive tray icon) is 10 MB/s maximum, whereas it is 60MB/s on the other mac/windows machines.

When uploading/downloading from web UI it is much faster and similar to other machines.

I don't know if the repetitive "Failed to open" line in `seadrive.log` is related to the issue but I have no other lead. Is there a way to enable some sort of debug mode for seadrive ?

**Comment by killing** [comment]
Well, for the slowly showing up issue, it may due to that you didn't look around in the SeaDrive folder. SeaDrive will only try to get library information update from the server after it detects file system activities in SeaDrive folder. Otherwise it assumes that the user is not interested in it at the moment. The purpose is to reduce the amount of requests sent to the server.

For the upload/download speed, it's normal for the Drive or sync client to have lower speed than the browser. It chunks large file into many a few MB blocks then send the blocks one by one. This will affect the transfer speed of TCP protocol. (TCP will gradually increase the speed after sending many bytes continuously.) The browser uploads the file in a single request, so it generally faster.

**Comment by monkeydri** [comment]
Yes it starts to sync only once you go in the folder, the thing is it takes a lot of time, I have another mac machine considerably less powerfull (old macbook) and on the same folders tested the newly added or updated files appear almost instantly and upload/download transfers are about 60MB/s.

**Comment by ydixt** [comment]
@killing Is there any way to debug the syncing process between seadrive and seafile server ? This slow syncing issue is encountered only on one mac machine.

I'm willing to take time to investigate this issue as it is really annoying (we already tried to uninstall re-install seadrive and perform a system cleanup.

**Comment by ydixt** [comment]
I created a separate issue (#134) for the slow syncing process on mac as we faced the same problem on other mac machines while not seeing the log message `file-cache-mgr.c(3236): Failed to open dir /Users/username/.seadrive/da...`

**Comment by avelo** [comment]
Same here, logs flooded with `file-cache-mgr.c(3236): Failed to open dir /Users/username/.seadrive/data/file-cache/root/`. 6 new lines written each 5 seconds. As a note, that dir doesn't exists, the existing one has a hash name instead of `root` word. Manually creating the directory doesn't help.

**Comment by killing** [comment]
This log has been removed in 1.0.8 version.
